examples/caching_middleware.clj

```diff
@@ -0,0 +1,54 @@
+(ns clj-http.examples.caching-middleware
+  (:require
+   [clj-http.client :as http]
+   [clojure.core.cache :as cache]))
+
+(def http-cache (atom (cache/ttl-cache-factory {} :ttl (* 60 60 1000))))
+
+(defn slurp-bytes
+  "Slurp the bytes from a slurpable thing"
+  [x]
+  (with-open [out (java.io.ByteArrayOutputStream.)]
```

Comment:
Earlier when using slurp on it's own there were exceptions thrown when the response body was an gzipped http stream, which lead to a path of avoiding slurp altogether... I've used your suggestion and it is fine 👍